cmake/Modules/Platform/Emscripten.cmake

```diff
@@ -17,6 +17,9 @@ set(CMAKE_SYSTEM_VERSION 1)
 
 set(CMAKE_CROSSCOMPILING TRUE)
 
+# Don't do compiler autodetection by default, but leave option for frameworks that require it
+option(EMSCRIPTEN_FORCE_COMPILERS "Force C/C++ compiler" ON)
```

Comment:
Also, let's be daring and default the option to `OFF` state. Tested this in off state locally on OS X and Windows with latest CMake 3.6, and it works properly. Having the option to force old behavior so that people have a way to work around if this causes a regression is a good idea.